src/censys/cloud_connectors/gcp_connector/connector.py

```diff
@@ -104,189 +107,402 @@ def scan_all(self):
+                project = ResourceSearchResult.to_dict(result)
+            except json.decoder.JSONDecodeError:  # pragma: no cover
+                self.logger.debug(f"Failed to parse project: {project}")
+                continue
+            try:
+                if versioned_resource := self.check_asset_version(
+                    GcpCloudAssetInventoryTypes.PROJECT, project
+                ):
+                    resource = versioned_resource["resource"]
+                    version = versioned_resource["version"]
+                    if version == "v3":
+                        project_id = resource["projectId"]
+                        project_number = self.parse_project_number(resource["name"])
+                        name = resource.get("displayName", "")  # Optional
+                    else:
+                        project_id = resource["projectId"]
+                        project_number = resource["projectNumber"]
+                        name = resource.get("name", "")  # Optional
+                    if (not project_id) or (not project_number):
+                        self.logger.debug(f"Failed to parse project: {project}")
+                    projects[project_number] = {"project_id": project_id, "name": name}
+            except (
+                json.decoder.JSONDecodeError,
+                ValueError,
+                KeyError,
+            ) as e:  # pragma: no cover
+                self.logger.debug(f"Failed to parse project: {project}: {e}")
+                continue
+
+        self.logger.debug(f"Found {len(projects)} projects.")
+        return projects
+
+    def parse_project_number(self, path: str) -> str:
+        """Parses an asset name to extract the project number.
+
+        Args:
+            path (str): Project path in the format "projects/{PROJECT NUMBER}".
+
+        Returns:
+            str: Asset parts.
+        """
+        project = AssetServiceClient.parse_common_project_path(path)
+        if project_number := project.get("project"):
+            return project_number
+        return ""
+
+    def return_if_str(self, val: Any) -> str:
+        """Return the value if it is a string.
+
+        Args:
+            val (Any): The value to check.
+
+        Returns:
+            str: The value if it is a string.
+        """
+        if isinstance(val, str):
+            return val
+        return ""
+
+    def check_asset_version(
+        self, asset_type: GcpCloudAssetInventoryTypes, asset: dict
+    ) -> Optional[dict]:
+        """Check if the asset version is supported and returns the resource if it is.
+
+        Args:
+            asset_type (GcpCloudAssetInventoryTypes): Asset type.
+            asset (dict): Asset.
+
+        Returns:
+            Optional[dict]: Resource with supported version, if it exists.
+        """
+        versioned_resources = asset.get("versioned_resources", [])
+        # Found resource with version `v1`
+        if len(versioned_resources) == 1:
```

Comment:
Can you introduce some locals so this is easier to follow?

count = len(versioned_resources)
supported = GcpApiVersions.SUPPORTED_VERSIONS.get_versions(asset_type)
unsupported = GcpApiVersions.UNSUPPORTED_VERSIONS.get_versions(asset_type)

It might also be worth making a parse_version function so the blocks aren't duplicated.